### ISSUE TITLE
Question: How is this package used?

### ISSUE THREAD

**Issue by spiderkeys** [ulricheck/conan-cuda_dev_config#1]
Hi @ulricheck 

I've been working on some camera streaming applications using NvPipe and used your conan-nvpipe recipe to bootstrap the efforts to get everything moved over to a conan workflow.

I noticed in conan-nvpipe that this repo is used as a dependency, however I can't really tell how it is being used. Also, every time it run 'conan install' I get something that looks like this:

```
cuda_dev_config/1.0@spiderkeys/test: running command: "/usr/local/cuda/bin/nvcc" --version
cuda_dev_config/1.0@spiderkeys/test: Invalid response from calling nvcc --version: 'nvcc: NVIDIA (R) Cuda compiler driver\nCopyright (c) 2005-2019 NVIDIA Corporation\nBuilt on Wed_Apr_24_19:10:27_PDT_2019\nCuda compilation tools, release 10.1, V10.1.168'
```
This could be because I've updated everything to use the new version of NvPipe which depends on Nvidia Codec SDK 9.0.20 and Cuda 10, but I thought I would check to see what your intentions were. 

Thanks!

**Comment by ulricheck** [comment]
Dear @spiderkeys 

This package was inspired by the python_dev_config package from bincrafters (https://github.com/bincrafters/conan-python_dev_config) to allow package artifacts to depend on a specific CUDA version it was compiled against.

It has not been tested with multiple versions of CUDA SDK so far - so improvements are very welcome.

the basic idea is to set the package attribute  "cuda_version" to the version number of the CUDA SDK the package was installed with, therefore - if you download binary artifacts with a different CUDA version, it would require you to recompile the dependent packages.

I saw that CUDA Version 10.1 is not listed in conanfile, which could be fixed easily .. however i'm not sure how what exactly is the reason for your error.

**Comment by spiderkeys** [comment]
@ulricheck Ah, I think I understand now.

To clarify, if I wanted to build three versions of NvPipe, using CUDA 8, 9, and 10, I would build three versions of cuda_dev_config, each building using a different docker image containing those respective versions (if using a docker workflow) and the cuda_version option will be detected and set automatically for each. 

Then I would build the three versions of NvPipe (or any CUDA application/library) pulling in cuda_dev_config and then using the cuda_version option to specify which cuda_dev_config package version to use. If I used this approach for all CUDA libraries/applications I built, if I had multiple libraries being pulled in via Conan, this would detect conflicts (or rather the lack of a package with this option) in case one or more of them was built using a different version of CUDA that didn't match the cuda_version option I specified.

Do I understand correctly?

**Comment by spiderkeys** [comment]
Re: the invalid response from NVCC, I think that was due to the cuda_version() regex match line using a hardcoded '10.0' version, rather than using self.options.cuda_version.

When I change to this:
```python
match = re.match( r".*, (\w+) ({}).*".format( self.options.cuda_version ), result.splitlines()[-1])
```
I get the correct response:
```
cuda_dev_config/1.0@spiderkeys/test: running command: "/usr/local/cuda/bin/nvcc" --version
cuda_dev_config/1.0@spiderkeys/test: Found CUDA SDK: 10.1
```

**Comment by ulricheck** [comment]
Yes .. i saw that the version was hard-coded in the initial version, which was not intended .. i'll fix this in a future update.
